server/src/main/java/org/exoplatform/chat/services/mongodb/UserServiceImpl.java

```diff
@@ -493,6 +504,121 @@ private SpaceBean getSpace(String roomId, String dbName)
     return users;
   }
 
+  @Override
+  public List<UserBean> getSuggestionUsers(String currentUserName, String roomId, String filter, String dbName, int limit) {
+    Map<String, UserBean> users = new LinkedHashMap<>();
+
+    String originalFilter = filter;
+    filter = filter.replaceAll(" ", ".*");
+    Pattern regex = Pattern.compile(filter, Pattern.CASE_INSENSITIVE);
+
+    BasicDBObject un = new BasicDBObject("user", regex);
+    BasicDBObject fn = new BasicDBObject("fullname", regex);
+    List<BasicDBObject> orList = new ArrayList<BasicDBObject>();
+    orList.add(un);
+    orList.add(fn);
+    BasicDBObject filterQuery = new BasicDBObject("$or", orList);
+
+    // Search in team members first
+    if (roomId != null && !"".equals(roomId)) {
+      // Search in room first
+      //removing "space-" prefix
+      if (roomId.indexOf(ChatService.SPACE_PREFIX)==0) {
+        roomId = roomId.substring(ChatService.SPACE_PREFIX.length());
+      }
+      //removing "team-" prefix
+      if (roomId.indexOf(ChatService.TEAM_PREFIX)==0) {
+        roomId = roomId.substring(ChatService.TEAM_PREFIX.length());
+      }
+
+      BasicDBObject spaces = new BasicDBObject("spaces", roomId);
+      BasicDBObject teams = new BasicDBObject("teams", roomId);
+      orList = new ArrayList<BasicDBObject>();
+      orList.add(spaces);
+      orList.add(teams);
+      BasicDBObject teamQuery = new BasicDBObject("$or", orList);
+
+      BasicDBObject query = new BasicDBObject("$and", Arrays.asList(teamQuery, filterQuery));
+      searchByQuery(users, dbName, query, limit);
+    }
+
+    int remain = limit - users.size();
+
+    // Then search in connection
+    if (remain > 0 && currentUserName != null && !"".equals(currentUserName)) {
+      // We can not inject IdentityManager to this service here because it threw NPE,
+      // maybe Guice container only return this service instance when portal container started
+      IdentityManager identityManager = GuiceManager.getInstance().getInstance(IdentityManager.class);
```

Comment:
How did you try to inject it ?
I think we should open a dedicated issue if there is no way to inject a service cleanly. It is ugly to have to use this solution.